src/main.ts

```diff
@@ -249,6 +250,35 @@ function handleColorRampClick({ target }: MouseEvent): void {
   canv.render();
 }
 
+function getFeatureValue(id: number): number {
+  if (!featureName || !dataset) {
+    return -1;
+  }
+  // Look up feature value from id
+  return dataset.getFeatureData(featureName)?.data[id] || -1;
+}
+
+function onMouseMove(event: MouseEvent): void {
+  if (!dataset) {
+    return;
+  }
+  const id = canv.getIdAtPixel(event.offsetX, event.offsetY);
+  if (id === -1) {
+    // Ignore background pixels
+    return;
```

Comment:
does this return mean that you don't clear out the last feature value when you move from foreground to background?

Reply:
Yeah, it keeps the last hovered feature unless the cursor exits the canvas (there's a video preview of it in the PR). I found that if I reset the hovered value on touching background pixels the hovered feature value would jump around a lot, especially given how small some of the cell targets are.